src/state/lottery/helpers.ts

```diff
@@ -0,0 +1,256 @@
+import BigNumber from 'bignumber.js'
+import { request, gql } from 'graphql-request'
+import { ethers } from 'ethers'
+import { GRAPH_API_LOTTERY } from 'config/constants/endpoints'
+import { LotteryStatus, LotteryTicket } from 'config/constants/types'
+import lotteryV2Abi from 'config/abi/lotteryV2.json'
+import { getLotteryV2Address } from 'utils/addressHelpers'
+import { multicallv2 } from 'utils/multicall'
+import {
+  LotteryUserGraphEntity,
+  LotteryRoundGraphEntity,
+  LotteryRound,
+  UserTicketsResponse,
+  UserRound,
+  LotteryRoundUserTickets,
+} from 'state/types'
+import { getLotteryV2Contract } from 'utils/contractHelpers'
+
+const lotteryContract = getLotteryV2Contract()
+
+export const fetchLottery = async (lotteryId: string): Promise<LotteryRound> => {
+  try {
+    const lotteryData = await lotteryContract.viewLottery(lotteryId)
+    const {
+      status,
+      startTime,
+      endTime,
+      priceTicketInCake,
+      discountDivisor,
+      treasuryFee,
+      firstTicketId,
+      lastTicketId,
+      amountCollectedInCake,
+      finalNumber,
+      cakePerBracket,
+      countWinnersPerBracket,
+      rewardsBreakdown,
+    } = lotteryData
+
+    const statusKey = Object.keys(LotteryStatus)[status]
+    const serializedCakePerBracket = cakePerBracket.map((cakeInBracket) => {
+      return new BigNumber(cakeInBracket.toString()).toJSON()
+    })
+    const serializedCountWinnersPerBracket = countWinnersPerBracket.map((winnersInBracket) => {
+      return new BigNumber(winnersInBracket.toString()).toJSON()
+    })
+    const serializedRewardsBreakdown = rewardsBreakdown.map((reward) => {
+      return new BigNumber(reward.toString()).toJSON()
+    })
```

Comment:
Yea man good point. I've added two here - https://github.com/pancakeswap/pancake-frontend/blob/d68eed053148fc6dd45f733c67395f1ae7b98883/src/utils/bigNumber.ts#L10-L13

I would break theses into a separate PR but we're a bit tight on time now to be merging multiple different pieces.